File: packages/ruby-sdk/lib/strait/operations/secrets.rb
# frozen_string_literal: true

module Strait
  module Operations
    # Service for secret management.
    class SecretsService < BaseService
      # List all secrets.
      # GET /v1/secrets
      def list(query: nil)
        _request(:get, "/v1/secrets", query: query)
      end

      # Create a new secret.
      # POST /v1/secrets
      def create(body)
        _request(:post, "/v1/secrets", body: body)
      end

      # Delete a secret.
      # DELETE /v1/secrets/{secretID}
      def delete(secret_id)
        _request(:delete, "/v1/secrets/{secretID}", path_params: { "secretID" => secret_id })
        nil
      end
    end
  end
end

File: packages/python-sdk/strait/operations/secrets.py
"""Secret management operations."""

from __future__ import annotations

from typing import Any

from strait.operations._base import AsyncBaseService, BaseService


class SecretsService(BaseService):
    def list(self, *, query: dict[str, str] | None = None) -> dict[str, Any]:
        return self._request("GET", "/v1/secrets", query=query)

    def create(self, body: Any) -> dict[str, Any]:
        return self._request("POST", "/v1/secrets", body=body)

    def delete(self, secret_id: str) -> dict[str, Any]:
        return self._request(
            "DELETE", "/v1/secrets/{secretID}", path_params={"secretID": secret_id},
        )


class AsyncSecretsService(AsyncBaseService):
    async def list(self, *, query: dict[str, str] | None = None) -> dict[str, Any]:
        return await self._request("GET", "/v1/secrets", query=query)

    async def create(self, body: Any) -> dict[str, Any]:
        return await self._request("POST", "/v1/secrets", body=body)

    async def delete(self, secret_id: str) -> dict[str, Any]:
        return await self._request(
            "DELETE", "/v1/secrets/{secretID}", path_params={"secretID": secret_id},
        )
